src/test/java/com/hazelcast/wm/test/DeferredWriteClusterTest.java

```diff
@@ -139,6 +139,18 @@ public void test_setThenGetAttribute() throws Exception {
         assertEquals("value", executeRequest("setGet", serverPort1, cookieStore));
     }
 
+    @Test(timeout = 20000)
+    public void testTransientAttributeDeferredWrite() throws Exception {
```

Comment:
This test needs to fail before any changes are made to the production classes but it passed even when no changes are made to `HazelcastHttpSession`. Can you please write a test that fails in the absence of your fix and passes with your fix so that we can get a better understanding of the problem?